zookeeper-server/src/main/java/org/apache/zookeeper/server/NettyServerCnxnFactory.java

```diff
@@ -378,8 +411,33 @@ public void operationComplete(Future<Channel> future) {
             }
         }
     }
+
+    @Sharable
+    static class ReadIssuedTrackingHandler extends ChannelDuplexHandler {
+
```

Comment:
nit: this should be a singleton instance...

Reply:
@normanmaurer I'm not sure I understand your suggestion about the singleton here, can you give more detail or example?

Reply:
You can reuse the same instance of this class across different channels. Just create one instance and reuse it

Reply:
That's what it is doing now, in the current code, there is only a single instance.

Reply:
I was more talking about something like this:

```
static class ReadIssuedTrackingHandler extends ChannelDuplexHandler {
    static final ReadIssuedTrackingHandler INSTANCE = new ReadIssuedTrackingHandler();
    private ReadIssuedTrackingHandler() { }
   
   ...
}
```

And then use `ReadIssuedTrackingHandler.INSTANCE` when you add it to the pipeline. This way you only need one instance per JVM... That said it is basically just a nit and if it matters depends on how often you create new clients / servers etc. So feel free to ignore.

Reply:
I like this approach. What do you think @lvfangmin ?

Reply:
Currently, there is only a single instance of ReadIssuedTrackingHandler created in NettyServerCnxnFactory, and there is only a single instance of NettyServerCnxnFactory per JVM lifecycle, so ReadIssuedTrackingHandler is only 1 instance per JVM.

Personally, I don't see there is value to create a static instance in ReadIssuedTrackingHandler, I'll keep the current implementation here.